### PR TITLE
Update CDASimAdapter to accomadate changes in SensorConfig file

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
### Background (Related changes)
This PR was created to accommodate changes in the Sensor Configuration file introduced by this PR (https://github.com/usdot-fhwa-stol/carma-streets/pull/427). To summarize, the sensor configuration file format was changed to allow for reference location to be specified in WGS84 coordinates or cartesian coordinates. For Simulation cartesian coordinates work well since many simulation environments do not have real-world equivalents and cartesian coordinates are easy to pull from different map editors and viewers like [JSOM](https://josm.openstreetmap.de/). For real world deployments however WGS84 coordinates are far easier to use since using cartesian coordinates introduces a dependency on an lanlet2 map for the area. It is far easier to use built in or separate GPS devices to get a reference location for a mounted sensor.
#### Old Format of Sensor Configuration JSON:
```json
[
  {
    "sensorId": "SomeID",
    "type": "SemanticLidar",
    "location": {
      "x": 1.0, /* in meters */
      "y": 2.0, /* in meters */
      "z": -3.2 /* in meters */
     },
     "orientation": {
       "yaw": 0.0,
       "pitch": 0.0,
       "roll": 0.0
      }
  }
]
```  
#### New Format of Sensor Configuration JSON:
```json
[
  {
    "sensorId": "sensor_1",
    "type": "SemanticLidar",
    "ref": {
      "type": "CARTESIAN",
      "location": {
        "x": 1.0, /* in meters */
        "y": 2.0, /* in meters */
        "z": -3.2 /* in meters */
      },
       "orientation": { 
        "yaw": 0.0,
        "pitch": 0.0,
        "roll": 0.0
        }
    }
  } 
]
```

### Changes

This change in Sensor Configuration file means that we need to change how we read and populate the registration message that goes out from the CDASim Adapter Plugin to CDASim. The content of this registration remains unchanged, we just need to read this information from the new `ref` element in the Sensor Configuration JSON and populate the existing registration. Lastly, we added logic for the registration to ignore any sensor with a WGS84 reference location since we currently only have support 
```json
{
  "infrastructureId": "id_123",
  "location": {
    "x": 0,
    "y": 0,
    "z": 0
  },
  "rxMessageIpAddress": "127.0.0.1",
  "rxMessagePort": 8686,
  "sensors": {
    "sensorId": "sensor1",
    "type": "SemanticLidar",
    "location": {
      "x": 0.0,
      "y": 0.0,
      "z": 0.0
     },
     "orientation": {
       "yaw": 0.0,
       "pitch": 0.0,
       "roll": 0.0
      }
  },
  "simulatedInteractionPort": 7576,
  "timeSyncPort": 7575
}

```

## Related Issue
[FCP-37
](https://usdot-carma.atlassian.net/browse/FCP-37)

## Motivation and Context
See description

## How Has This Been Tested?
Local Integration Testing and unit testing

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp [hunk 1]
```diff
@@ -261,8 +261,63 @@ namespace CDASimAdapter{
...
+                    "yaw": 0.0,
+                    "pitch": 0.0,
+                    "roll": 0.0
+                    }
+            }
+        } 
+        ] */
+        auto sensor_configuration = read_json_file(file_path);
+        Json::Value sensors_registration;
+        for ( int index = 0; index < sensor_configuration.size(); ++index ) {
```
Comment (dan-du-car) [inline src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp:291]: Is this sensor_configuration JSON::value possibly a array? Why do we event need a for loop if it is just a JSON object?

File: src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp [hunk 2]
```diff
@@ -261,8 +261,63 @@ namespace CDASimAdapter{
...
+            Json::Value sensor;
+            sensor["sensorId"] = sensor_configuration[index]["sensorId"];
```
Comment (dan-du-car) [inline src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp:293]: What happens if sensorId is missing in the JSON file?

File: src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp (hunk 1 above)
Comment (paulbourelly999) [reply src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp:291]: It is an array. See the sample file. The SDSM service can only support on sensor for each instance but V2X-Hub and CDASim can create multiple sensors for a single V2X-Hub instance.

File: src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp (hunk 2 above)
Comment (paulbourelly999) [reply src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp:293]: We get a warning that the file can not be found and we send a infrastructure registration without sensor information

**Review by dan-du-car — APPROVED** [review]

**Pull request merged**